app/src/main/java/com/google/maps/android/ktx/demo/MainActivity.kt

```diff
@@ -0,0 +1,129 @@
+import com.google.maps.android.collections.MarkerManager
+import com.google.maps.android.collections.PolygonManager
+import com.google.maps.android.collections.PolylineManager
+import com.google.maps.android.data.geojson.GeoJsonLayer
+import com.google.maps.android.data.geojson.GeoJsonLineStringStyle
+import com.google.maps.android.data.geojson.GeoJsonPolygonStyle
+import com.google.maps.android.ktx.MapsExperimentalFeature
+import com.google.maps.android.ktx.awaitMap
+import com.google.maps.android.ktx.demo.model.MyItem
+import org.json.JSONException
+
+/**
+ * A demo of multiple layers on the map.
+ *
+ * To add your Maps API key to this project:
+ *   1. Create a file app/gradle.properties
+ *   2. Add this line, where YOUR_API_KEY is your API key:
+ *        MAPS_API_KEY="YOUR_API_KEY"
+ */
+class MainActivity : AppCompatActivity() {
+    var mIsRestore: Boolean = false
+
+    @MapsExperimentalFeature
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        mIsRestore = savedInstanceState != null
+        setContentView(R.layout.activity_main)
+        val mapFragment = supportFragmentManager.findFragmentById(R.id.map) as? SupportMapFragment
+        lifecycle.coroutineScope.launchWhenCreated {
+            check(mapFragment != null)
+            val googleMap = mapFragment.awaitMap()
+            if (!mIsRestore) {
+                googleMap.moveCamera(CameraUpdateFactory.newLatLngZoom(LatLng(51.403186, -0.126446),10F))
+            }
+            showMapLayers(googleMap)
+        }
+    }
+
+    private fun showMapLayers(map: GoogleMap) {
+        // Shared object managers - used to support multiple layer types on the map simultaneously
+        val markerManager = MarkerManager(map)
+        val groundOverlayManager = GroundOverlayManager(map)
+        val polygonManager = PolygonManager(map)
+        val polylineManager = PolylineManager(map)
+
+        addClusters(map, markerManager)
+        addGeoJson(map, markerManager, polylineManager, polygonManager, groundOverlayManager)
+    }
+
+    private fun addClusters(map: GoogleMap, markerManager: MarkerManager) {
+        val clusterManager = ClusterManager<MyItem>(this, map, markerManager)
+        map.setOnCameraIdleListener(clusterManager)
+
+        try {
+            val items = MyItemReader().read(resources.openRawResource(R.raw.radar_search))
+            clusterManager.addItems(items)
+        } catch (e: JSONException) {
+            Toast.makeText(this, "Problem reading list of markers.", Toast.LENGTH_LONG).show()
+            e.printStackTrace()
+        }
+    }
+
+    private fun addGeoJson(map: GoogleMap, markerManager: MarkerManager, polylineManager: PolylineManager, polygonManager: PolygonManager, groundOverlayManager: GroundOverlayManager) {
+        // GeoJSON polyline
+        // FIXME - Why can't we use named parameters here?
+        val geoJsonLineLayer = GeoJsonLayer(
```

Comment:
@barbeau named parameters aren't working here because this needs to be lower-cased (i.e. `geoJsonLayer(map=map,...)`. This fake constructor approach is a pattern across other KTX libraries since you can't truly create another `init` method through an extension.

Reply:
Ah, ok, thanks! I think the README needs to be updated then?

https://github.com/googlemaps/android-maps-ktx#named-parameters-and-default-arguments

It has:

```kotlin
val layer = GeoJsonLayer(...
```

Reply:
Ah, yes. It should be updated as well.

Reply:
Just opened https://github.com/googlemaps/android-maps-ktx/pull/30 for this